### PR TITLE
Remove wordpress-external-dependencies-plugin

### PR DESCRIPTION
**Pull request by sirreal**
#34122 transitions to canonical package.
This package is no longer needed.

### CODE REVIEW

**Comment by ockham** [comment]
Pushed a commit to update `calypso-build`'s README to reference to `@wordpress/dependency-extraction-webpack-plugin` rather than `wordpress-external-dependencies-plugin`, which seems to have fallen through the cracks during #34122. This should remove the last remaining reference to the old Webpack plugin name.

**Comment by ockham** [comment]
BTW we don't want/need a final version to include some sort of note for deprecation of the npm package? We can do the solely through [`npm deprecate`](https://docs.npmjs.com/cli/deprecate), right?

**Review by ockham — APPROVED** [review]
:+1: 

(I added a commit, but it's a search-replace job on a README and one Changelog line only)

File: packages/calypso-build/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # [Unreleased]
 
+- Update `@automattic/wordpress-external-dependencies-plugin` references in `README.md` to `@wordpress/dependency-extraction-webpack-plugin`.
```
Comment (sirreal) [inline packages/calypso-build/CHANGELOG.md]: This was part of 3.0.0, look a few lines lower 🙂

File: packages/calypso-build/README.md [hunk 2]
```diff
@@ -64,7 +64,7 @@ That `webpack.config.js` introduces one rather WordPress/Gutenberg specific "env
 
 The impact of this option is twofold:
 
-1. It will make Webpack use `wordpress-external-dependencies-plugin` to infer NPM packages that are commonly used by Gutenberg blocks (anything in the `@wordpress/` scope, `lodash`, React, jQuery, etc) from the source files it bundles, and produce a `.deps.json` file containing an array of those dependencies for use with `wp_enqueue_script`. For more information, see `wordpress-external-dependencies-plugin`'s [docs](../wordpress-external-dependencies-plugin/README.md).
+1. It will make Webpack use `@wordpress/dependency-extraction-webpack-plugin` to infer NPM packages that are commonly used by Gutenberg blocks (anything in the `@wordpress/` scope, `lodash`, React, jQuery, etc) from the source files it bundles, and produce a `.deps.json` file containing an array of those dependencies for use with `wp_enqueue_script`. For more information, see `@wordpress/dependency-extraction-webpack-plugin`'s [docs](https://developer.wordpress.org/block-editor/packages/packages-dependency-extraction-webpack-plugin/).
```
Comment (sirreal) [inline packages/calypso-build/README.md]: Thanks!

File: packages/calypso-build/CHANGELOG.md (hunk 1 above)
Comment (ockham) [reply packages/calypso-build/CHANGELOG.md]: Not quite :grimacing: I'm specifically calling out the changes to the README here -- which weren't in 3.0.0 :wink: 
(But it's debatable whether they're worth mentioning in the Changelog -- feel free to remove.)

**Pull request merged**